server/middleware/users.coffee

```diff
@@ -405,3 +406,34 @@ module.exports =
     ids = rows.map (r) -> mongoose.Types.ObjectId(r.mongoid)
     connection.end()
     return ids    
+
+  resetProgress: wrap (req, res) ->
+    unless req.user
+      throw new errors.Unauthorized()
+    if req.params.handle isnt req.user.id
+      throw new errors.Forbidden('Users may only delete themselves')
+    if req.user.isAdmin()
+      throw new errors.Forbidden('Admins cannot reset progress')
+    yield [
+      LevelSession.remove({creator: req.user.id})
+      EarnedAchievement.remove({user: req.user.id})
+      UserPollsRecord.remove({user: req.user.id})
```

Comment:
I'm not sure why we would reset this.  Its not like their answers are likely to change.